### PR TITLE
Fix 7645: Enable remote debugging by default and proxy devtools request

### PR DESCRIPTION
**Pull request by jumde**
Resolves https://github.com/brave/brave-browser/issues/7645

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: chromium_src/chrome/browser/devtools/url_constants.cc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "chrome/browser/devtools/url_constants.h"
+
+const char kRemoteFrontendDomain[] = "devtools.brave.com";
```
Comment (bridiver) [inline chromium_src/chrome/browser/devtools/url_constants.cc:8]: can we not use the normal static redirect? My concern is that we're changing urls in different places and that makes it harder to find how/where/what we're changing

Comment (jumde) [reply chromium_src/chrome/browser/devtools/url_constants.cc:8]: @bridiver - I was trying that first, but it requires a much bigger re-write of the code. 

Currently, the brave-proxy code lives in `brave_proxying_url_loader_factory.cc` which lives in `//brave/browser/net`. To use static redirect we need to invoke `MaybeProxyRequest` from `devtools_instrumentation::WillCreateURLLoaderFactory` which is present in `//content/browser`. 

So for static redirect to work, we'll need to move the proxying code to `//content/` or `//net`.

**Comment by bsclifton** [comment]
Changes LGTM - when running locally on Windows, it starts working and then gets to the generating report part. Unfortunately, that seems to close dev tools? Not sure if something is not being proxied properly

**Comment by bsclifton** [comment]
As mentioned in the issue, I believe what I ran into is captured w/ https://bugs.chromium.org/p/chromium/issues/detail?id=1036525

File: chromium_src/chrome/browser/devtools/url_constants.cc (hunk 1 above)
Comment (iefremov) [reply chromium_src/chrome/browser/devtools/url_constants.cc:8]: @jumde do you remember if we have an issue for moving proxying code to `//net`? At least `brave_system_request_handler.cc`. I guess this is not the first layering problem with this

Comment (jumde) [reply chromium_src/chrome/browser/devtools/url_constants.cc:8]: @iefremov - Filed one: https://github.com/brave/brave-browser/issues/9770 - Thanks for the reminder.

**Pull request merged**